rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -2402,6 +2402,58 @@ def test_upgrade_db_30_to_31(user_data_dir, db_with_set_version):  # pylint: dis
     ]
 
 
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_upgrade_db_31_to_32(user_data_dir):  # pylint: disable=unused-argument  # noqa: E501
+    """Test upgrading the DB from version 31 to version 32.
+
+    Test that the ids at the history_events table are correctly updated after
+    changing how they are generated
+
+    Check that subtype is correctly updated
+    """
+    msg_aggregator = MessagesAggregator()
+    _use_prepared_db(user_data_dir, 'v31_rotkehlchen.db')
+    db = _init_db_with_target_version(
+        target_version=31,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db.conn.cursor()
+    result = cursor.execute(
+        'SELECT identifier from history_events',
+    )
+    old_ids = result.fetchall()
+    assert len(old_ids) == 3
+    cursor.execute(
+        'SELECT subtype from history_events',
+    )
+    subtypes = [row[0] for row in cursor.fetchall()]
+    assert set(subtypes) == {'staking deposit asset', 'staking receive asset', None}
+
+    # Execute migration
+    db = _init_db_with_target_version(
+        target_version=32,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db.conn.cursor()
+    # Finally also make sure that we have updated to the target version
+    assert db.get_version() == 32
+    result = cursor.execute(
+        'SELECT identifier from history_events',
+    )
+    new_ids = result.fetchall()
+    assert len(new_ids) == 3
+    for new_id in new_ids:
+        assert new_id not in old_ids
+
+    cursor.execute(
+        'SELECT subtype from history_events',
+    )
+    subtypes = [row[0] for row in cursor.fetchall()]
+    assert set(subtypes) == {'staking deposit asset', 'staking receive asset', 'reward'}
```

Comment:
Shouldn't you also assert on the identifier change? (unless we change the way identifiers are created)